### PR TITLE
remove unnecessary clones

### PR DESCRIPTION
**Pull request by mohad12211**
The Rust Compiler thinks that the closure will always execute and move the `remove_cookie_on_fail` because it can't figure out that the rest of the function and the closure are mutually exclusive.

Maybe because the use of `map_err` with `?` involves some kind of logic, so I guess it's hard to know in compile time that those two when combined together, will produce two mutually exclusive branches.

The solution is to use a match to show explicitly that there are two mutually exclusive branches. So we can safely move the `remove_cookie_on_fail` to the Error branch.

### CODE REVIEW

File: src/auth.rs [hunk 1]
```diff
@@ -94,21 +94,23 @@ impl<const USER_ROLE: u32> FromRequestParts<AppState> for AuthExtractor<USER_ROL
...
+        let session_id = match cookies.private(key).get(SESSION_COOKIE_NAME) {
+            None => {
                 tracing::error!("auth-extractor: failed to get session_id cookie");
-                cookies.remove(remove_cookie_on_fail.clone().finish());
-                AuthError::InvalidSession
-            })?;
+                cookies.remove(remove_cookie_on_fail.finish());
+                return Err(AuthError::InvalidSession);
+            }
+            Some(id) => id,
```
Comment (BKSalman) [inline src/auth.rs:103]: Indentation here is a little weird

**Review by BKSalman — CHANGES_REQUESTED** [review]

File: src/auth.rs (hunk 1 above)
Comment (mohad12211) [reply src/auth.rs:103]: what's weird about it? I ran `cargo fmt` on it and it produced the same formatting.

Comment (BKSalman) [reply src/auth.rs:103]: Will check it on my laptop

**Comment by BKSalman** [comment]
yeah nvm was half asleep

**Pull request merged**